include/deal.II/matrix_free/hanging_nodes_internal.h

```diff
@@ -376,6 +376,28 @@ namespace internal
     inline void
     HangingNodes<3>::setup_line_to_cell(const Triangulation<3> &triangulation)
     {
+      // In case all our cells are on the same level, there is nothing to
+      // do. We only need to check for active cells (non-artificial) that are
+      // on a level coarser than the finest one.
+      bool all_cells_on_same_level = true;
```

Comment:
Why not use `Triangulation::has_hanging_nodes()`?

Reply:
I don't want to go through the communication involved there, and wanted it to be a local operation.

Reply:
Wouldn't it be better to move the information to `number_cache`?

Reply:
At some point yes, but for the current release it feels too unsafe to touch that part. But let me use a similar code as the `has_hanging_nodes` function, to reduce the variability we have in the code base.

Reply:
I now pushed the update.